contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,491 @@
+    }
+
+    /**
+        @notice Deposit additional sOHM, wrap to gOHM, and update deposit record
+        @param depositId_ Deposit ID to direct additional gOHM to
+        @param amount_ Amount of new sOHM debt issued from donor to recipient
+    */
+    function addToSohmDeposit(uint256 depositId_, uint256 amount_) external {
+        if (isInvalidUpdate(depositId_, amount_)) revert YieldDirector_InvalidUpdate();
+        if (depositInfo[depositId_].depositor != msg.sender) revert YieldDirector_NotYourDeposit();
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        uint256 gohmAmount = staking.wrap(address(this), amount_);
+
+        _increaseDeposit(depositId_, gohmAmount);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault
+        @param depositId_ Deposit ID to remove gOHM deposit from
+        @param amount_ Amount of gOHM deposit to remove and return to donor
+    */
+    function withdrawPrincipal(uint256 depositId_, uint256 amount_) external {
+        uint256 amountWithdrawn = _withdraw(depositId_, amount_);
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault, and return it as sOHM
+        @param depositId_ Deposit ID to remove gOHM debt from
+        @param amount_ Amount of gOHM debt to remove and return to donor as sOHM
+    */
+    function withdrawPrincipalAsSohm(uint256 depositId_, uint256 amount_) external {
+        uint256 amountWithdrawn = _withdraw(depositId_, amount_);
+
+        IERC20(sOHM).approve(address(staking), amountWithdrawn);
+        staking.unwrap(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw all gOHM from all donor positions
+    */
+    function withdrawAll() external {
+        if (withdrawDisabled) revert YieldDirector_WithdrawalsDisabled();
+
+        uint256[] storage depositIds = depositorIds[msg.sender];
+
+        uint256 depositsLength = depositIds.length;
+        if (depositsLength == 0) revert YieldDirector_NoDeposits();
+
+        uint256 principalTotal = 0;
+
+        for (uint256 index = 0; index < depositsLength; ++index) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+
+            principalTotal += currDeposit.principalAmount;
+
+            _withdrawAllPrincipal(currDeposit.id);
+        }
+
+        IERC20(gOHM).safeTransfer(msg.sender, _toAgnostic(principalTotal));
+
+        emit AllWithdrawn(msg.sender, _toAgnostic(principalTotal));
+    }
+
+    /**
+        @notice Get deposited gOHM amounts for specific recipient (updated to current index
+                based on sOHM equivalent amount deposit)
+        @param donor_ Address of user donating yield
+        @param recipient_ Address of user receiving donated yield
+    */
+    function depositsTo(address donor_, address recipient_) external view returns (uint256) {
```

Comment:
would depositedTo be a better name? Or is frontend already used to depositsTo we don't want to change.